trojsten/submit/views.py

```diff
@@ -499,12 +505,96 @@ def task_submit_post(request, task_id, submit_type):
             return redirect(request.POST["redirect_to"])
         else:
             return redirect(reverse("task_submit_page", kwargs={"task_id": int(task_id)}))
-
+    elif submit_type == constants.SUBMIT_TYPE_TEXT:
+        return task_submit_post_susi(request, task_id, submit_type)
     else:
-        # Only Description and Source and Zip submitting is developed currently
+        # Only Description and Source and Zip and Text submitting is developed currently
         raise Http404
 
 
+@login_required
+def task_submit_post_susi(request, task_id, submit_type):
+    """Spracovanie uploadnuteho submitu pre susi"""
+    task = get_object_or_404(Task, pk=task_id)
+    form = TextSubmitForm(request.POST)
+
+    if form.is_valid():
+        now = timezone.now()
+        submitted_text = form.cleaned_data["submitted_text"]
+        solution = task.text_submit_solution.lower()
+        submitted_text = unidecode(submitted_text.replace(" ", "").lower())
+        if submitted_text == solution:
+            response = "OK"
+            points = constants.SUSI_POINTS_ALLOCATION["Full"]
+            if (
+                task.round.small_hint_date < now <= task.round.big_hint_date
+                and len(task.susi_small_hint) > 0
+            ):
+                points -= constants.SUSI_POINTS_ALLOCATION["Small Hint Deduction"]
```

Comment:
ja by som tieto susi specific veci ratal na urovni vysledkovky v rules nie pri submite.., textove submity so solutionami a auto-gradingom by mohli byt lahko znovu pouzitelne aj v inych sutaziach mimo susi..,

Reply:
Myslíš, že by to mohlo fungovať nejak tak, že v rules zadefinujeme funkcie ako napr. get_response_for_text_submit, a z toho to sa vo views zgeneruje všetko ostatné?

Reply:
pride mi to zbytocne..., 

tuna kludne mozes kazdemu submitu dat plny pocet bodov a potom ked pocitas vysledkovku spocitat kedy bol prvy spravny submit, kolko zlych bolo pred nim a podla toho vypocitat body ktore sa zobrazia vo vysledkovke..

tak isto funguje aj KSP doprogramovavanie...., ak pocas sutaze neodovzdas submit, ale pocas doprogramovavania naprogramujes spravne riesenie, v zozname submitov mas plny pocet bodov ale vo vysledkovke polovicny..

Reply:
No nam islo o to, aby to hned vyhodilo spravu, ze kolko presne bodov za tuto ulohu dostane clovek, lebo mame viac veci, za co sa odpocitavaju body z plneho poctu.